test/system/lib/run_metashell.cpp

```diff
@@ -19,21 +19,53 @@
 #include <metashell_system_test/system_test_config.hpp>
 #include <metashell_system_test/prompt.hpp>
 #include <metashell_system_test/raw_text.hpp>
-
-#include <just/process.hpp>
+#include <metashell_system_test/process_execution.hpp>
 
 #include <cassert>
 #include <sstream>
 #include <iostream>
+#include <cstdlib>
 
 using namespace metashell_system_test;
 
 namespace
 {
-  void pop_item(const std::string& item_, std::vector<std::string>& v_)
+  void run_metashell_assert(
+    const std::string& cond_,
+    bool cond_value_,
+    const process_execution& execution_,
+    const std::string& filename_,
+    int line_
+  )
+  {
+    if (!cond_value_)
+    {
+      const char sep[] = "----------------------------------";
+      std::cerr
+        << sep << std::endl
+        << "Assertion failed: " << cond_ << " at "
+        << filename_ << ":" << line_ << std::endl
+        << "Related child process execution:" << std::endl
+        << execution_
+        << sep << std::endl;
+      std::abort();
```

Comment:
`std::abort` seems a bit too aggressive. How about `std::exit(1)`, so the destructors can still execute?